### ISSUE TITLE
🐛 `noUndeclaredDependencies` doesn't report any diagnostics using `biome ci`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#2819]
### Environment information

```block
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "foot"
  JS_RUNTIME_VERSION:           "v21.7.3"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

```shell
❯ npm init --yes
❯ npm i -D @biomejs/biome
❯ cat biome.json
{
	"linter": {
		"rules": {
			"recommended": false,
			"nursery": {
				"noUndeclaredDependencies": "error"
			}
		}
	}
}

❯ cat index.js
import "a";

❯ npx biome ci index.js 
Checked 1 file in 475µs. No fixes needed.

❯ npx biome lint index.js 
index.js:1:8 lint/nursery/noUndeclaredDependencies ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The current dependency isn't specified in your package.json.
  
  > 1 │ import "a";
      │        ^^^
    2 │ 
  
  ℹ This could lead to errors.
  
  ℹ Add the dependency in your manifest.
  

Checked 1 file in 652µs. No fixes needed.
Found 1 error.
lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

Note: executing `biome check` yields the same result as `biome lint`.

### Expected result

`biome ci` should emit the same diagnostic as `biome lint`.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
